src/rules/no-if.ts

```diff
@@ -1,36 +1,54 @@
-import { getDocsUrl, isTestCase } from './util';
+import { TestCaseName, createRule, getNodeName, isTestCase } from './tsUtils';
+import {
+  AST_NODE_TYPES,
+  TSESTree,
+} from '@typescript-eslint/experimental-utils';
 
-const isTestArrowFunction = node =>
+const testCaseNames = new Set<string | null>([
+  ...Object.keys(TestCaseName),
+  'it.only',
+  'it.skip',
+  'test.only',
+  'test.skip',
+]);
+
+const isTestArrowFunction = (node: TSESTree.Node) =>
```

Comment:
This function is only used by `ArrowFunctionExpression`, so you can do `node: TSESTree.ArrowFunctionExpression` and ditch the `undefined` and `node.type` checks

Reply:
Actually apparently I can leave it as `TSESTree.Node`, but yeah much better!

Reply:
no need to widen the type when we don't need it 🙂

Reply:
Given the functions name (`isTestArrowFunction`), I think it should: either it should test that the type is `ArrowFunctionExpression`, or it should test if a given `ArrowFunctionExpression ` is a "test" arrow function.